cpu/mips_pic32_common/include/periph_cpu_common.h

```diff
@@ -43,10 +45,20 @@ extern "C" {
  */
 #define CPUID_LEN           (4U)
 
+#ifndef DOXYGEN
+/**
+ * @brief   Overwrite the default gpio_t type definition
+ * @{
+ */
+#define HAVE_GPIO_T
+typedef uint32_t gpio_t;
+/** @} */
+#endif
+
 /**
  * @brief   Override GPIO pin selection macro
  */
-#define GPIO_PIN(x,y)       ((x << 4) | (y & 0xf))
+#define GPIO_PIN(x, y)      (((_PORTB_BASE_ADDRESS & 0xFFFFF000) + (x << 8)) | y)
```

Comment:
~~I know it doesn't make a difference, but it would *feel* nicer if you used `_PORTA_BASE_ADDRESS` here.~~
Never mind, PIC32MX470F512H does not have PORTA.